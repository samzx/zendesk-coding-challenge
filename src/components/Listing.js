import React from 'react';

class Listing extends React.Component {

    handleClick = () => {
        this.props.setTicket(this.props.id);
    }
    render() {
        const {item, index} = this.props;
        return(
            <div className="listing" onClick={this.handleClick}>
                <div className="listing-subject">
                    <p>{item.subject}</p>
                </div>
                <div className="listing-time">
                    <p>{item.created_at}</p>
                </div>
                <div className="listing-priority">
                    <p>{item.priority}</p>
                </div>
                <div className="listing-type">
                    <p>{item.type}</p>
                </div>
            </div>
        );
    }
}

export default Listing;