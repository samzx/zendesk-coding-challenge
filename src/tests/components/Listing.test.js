import React from "react";
import { shallow } from "enzyme";
import Listing from "../../components/Listing";

import { response } from "../fixtures/response.fixtures";

describe("<Listing />", () => {
  it("should render Listing correctly", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render Listing correctly", () => {
    const wrapper = shallow(<Listing listing={response.tickets[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<Listing />'s priority display", () => {
  it("should convert urgent to 4 stars", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper.instance().visualPriority("urgent")).toEqual("★★★★");
  });
  it("should convert high to 3 stars", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper.instance().visualPriority("high")).toEqual("★★★");
  });
  it("should convert normal to 2 stars", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper.instance().visualPriority("normal")).toEqual("★★");
  });
  it("should convert low to 1 stars", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper.instance().visualPriority("low")).toEqual("★");
  });
  it("should convert unknown string to 0 stars", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper.instance().visualPriority("")).toEqual("");
  });
});

describe("<Listing />'s time display", () => {
  it("should shorten time to be easier to read given a valid date string", () => {
    const wrapper = shallow(<Listing />);
    expect(
      wrapper.instance().shortenTime("Thu Jun 07 2018 23:13:19 GMT+1000 (AEST)")
    ).toEqual("Thu Jun 07 2018 23:13:19");
  });
  it("should not show time given a invalid date string", () => {
    const wrapper = shallow(<Listing />);
    expect(wrapper.instance().shortenTime("abcdefg")).toEqual("");
  });
});

describe("<Listing />'s handleClick", () => {
  const mockSetTicket = props => {
    throw "ticket set";
  };
  it("should not set ticket if loading", () => {
    const wrapper = shallow(
      <Listing setTicket={mockSetTicket} loading={false} />
    );
    expect(() => {
      wrapper.instance().handleClick();
    }).toThrow("ticket set");
  });
  it("should set ticket if not loading", () => {
    const wrapper = shallow(
      <Listing setTicket={mockSetTicket} loading={true} />
    );
    expect(wrapper.instance().handleClick());
  });
});
