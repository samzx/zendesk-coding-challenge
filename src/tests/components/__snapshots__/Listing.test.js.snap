// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Listing /> should render Listing correctly 1`] = `""`;

exports[`<Listing /> should render Listing correctly 2`] = `
<div
  className="listing"
  onClick={[Function]}
>
  <div
    className="listing--left"
  >
    <div
      className="listing-subject"
    >
      <h3>
        commodo excepteur cupidatat aliquip quis
      </h3>
    </div>
    <div
      className="listing-time"
    >
      <p>
        Thu Jun 07 2018 23:13:13
      </p>
    </div>
  </div>
  <div
    className="listing--right"
  >
    <div
      className="listing-type"
    >
      <h3>
        question
      </h3>
    </div>
    <div
      className="listing-priority"
    >
      <p>
        ★★★★
      </p>
    </div>
  </div>
</div>
`;
